### DISCUSSION TITLE
Support For Generic RPCs

### DISCUSSION THREAD

**Discussion by Katerlad** [FirstGearGames/FishNet#472] [Feature Request]
Please add support for Generic RPCs! 

It would turn my code from this 😭 

```csharp
_player.UI.ShowView(nameof(LevelUpView));
```
To This 😍 

```csharp
   _player.UI.ShowView<LevelUpView>()
```
```csharp

List<View> _views = new();

[ObserversRpc]
public void ShowView<T>() where T : View
{
     foreach (View view in _views)
     {
         if (view.GetType == typeof(T))
         {
             view.Show();
         }
}

**Comment by FirstGearGames** [comment]
Not planned for a variety of codegen reasons. Plus run-time generics are problematic. Class generic, possible but still a ton of work.
I'm sure we'll get generic RPCs at some point but no time soon.
